src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -39,13 +40,19 @@
 
 (schema/defn new-rule :- Rule
   "Creates a new rule with an empty ACL"
-  ([type :- Type
-    pattern :- Pattern]
-    {:type type :path pattern :acl acl/empty-acl :method :any})
-  ([type :- Type
-    pattern :- Pattern
-    method :- Method]
-    {:type type :path pattern :acl acl/empty-acl :method method}))
+  [type :- Type
+   path :- schema/Str
+   method :- Method
+   sort-order :- schema/Int
+   name :- schema/Str]
+  {:type type
+   :path (condp = type
+           :path (re-pattern (str "^" (Pattern/quote path)))
+           :regex (re-pattern path))
+   :acl acl/empty-acl
+   :method method
+   :sort-order sort-order
+   :name name})
```

Comment:
This is a pretty significant refactoring I made.  I eliminated the constructor functions `new-path-rule` and `new-regex-rule`, which only provided the string path -> regex pattern conversion.

Now there's only one constructor, with a single arity - `new-rule`